website/static/js/koHelpers.js

```diff
@@ -436,6 +436,48 @@ ko.bindingHandlers.datePicker = {
 };
 
  /**
+ * Bind content of contenteditable to observable. Looks for maxlength attr
+ * and underMaxLength binding to limit input.
+ * Example:
+ * <div contenteditable="true" data-bind="editableHTML: <observable_name>" maxlength="500"></div>
+ */
+ko.bindingHandlers.editableHTML = {
+    init: function(element, valueAccessor, allBindings, bindingContext) {
+        var $element = $(element);
+        var initialValue = ko.utils.unwrapObservable(valueAccessor());
+        $element.html(initialValue);
+        $element.on('change input paste keyup blur', function() {
+            var observable = valueAccessor();
+            observable($element.html());
+        });
+    },
+    update: function(element, valueAccessor, allBindings, bindingContext) {
+        var $element = $(element);
+        var initialValue = ko.utils.unwrapObservable(valueAccessor());
+        var bindings = bindingContext;
+        var charLimit = $element.attr('maxlength');
+        if (charLimit && bindingContext.underMaxLength) {
+            var inputTextLength = $element[0].innerText.length;
+            var errorExceedsCharacterLimit = 'Exceeds character limit. Please reduce to ' + charLimit + ' characters or less.';
+            if (inputTextLength > parseInt(charLimit) + 1) {
+                bindingContext.underMaxLength(false);
+                if (bindingContext.errorMessage) {
+                    bindingContext.errorMessage(errorExceedsCharacterLimit);
+                }
+            } else {
+                bindingContext.underMaxLength(true);
+                if (bindingContext.errorMessage) {
+                    bindingContext.errorMessage('');
+                }
+            }
+        }
+        if (initialValue === '') {
+            $(element).html(initialValue);
```

Comment:
>
What I'm missing is, what resets the value to the empty string when I click outside of the comment box?
>

```
CommentModel.prototype.autosizeText = function(elm) {
    $(elm).find('textarea').autosize().focus();
    $(elm)
        .find('.atwho-input')
        .atwho(atjsConfig.at_config)
        .atwho(atjsConfig.plus_config)
        .bind('paste', atjsConfig.onPaste)
        .on('focusin', atjsConfig.lastElementBr)
        .on('focusout', atjsConfig.onlyElementBr)
        .keydown(function(e) {
            if(e.which === 13 && !e.isDefaultPrevented()) {
                atjsConfig.onReturn(e);
            }
        });
};
```

On focus out the `<br>` is removed which causes the observable and therefore the `initialValue` to be `''`.